### PR TITLE
[chore] add retry logic to flaky http test

### LINKED ISSUE #11654: TestTelemetryInit failing sporadically in GH Actions pipeline
**Describe the bug**
`TestTelemetryInit` in `service/telemetry/metrics_test.go` fails occasionally due to HTTP request refusing connection.

**Steps to reproduce**
Make a (draft) PR with any default change, and wait for unit tests to run.

**What did you expect to see?**
All tests passing (especially ones that aren't related to changes)

**What did you see instead?**
https://github.com/open-telemetry/opentelemetry-collector/actions/runs/11803774920/job/32882508723?pr=11653#step:6:558
**What version did you use?**
`main`

**Environment**
GH Actions

### PR DESCRIPTION
**Pull request by jackgopack4**
#### Description
attempts to fix flaky `service/telemetry/metrics` test

#### Link to tracking issue
Fixes #11654 

#### Testing
update to test helper function

#### Documentation
n/a

### CODE REVIEW

File: service/telemetry/metrics_test.go [hunk 1]
```diff
@@ -201,11 +202,31 @@ func createTestMetrics(t *testing.T, mp metric.MeterProvider) {
...
-	req, err := http.NewRequest(http.MethodGet, endpoint, nil)
+	client := &http.Client{
+		Timeout: 10 * time.Second,
+	}
+	var (
+		req *http.Request
+		err error
+		rr  *http.Response
+	)
+	req, err = http.NewRequest(http.MethodGet, endpoint, nil)
```
Comment (bogdandrutu) [inline service/telemetry/metrics_test.go:213]:
```suggestion
	req, err := http.NewRequest(http.MethodGet, endpoint, nil)
```

File: service/telemetry/metrics_test.go [hunk 2]
```diff
@@ -201,11 +202,31 @@ func createTestMetrics(t *testing.T, mp metric.MeterProvider) {
...
 	require.NoError(t, err)
 
-	rr, err := http.DefaultClient.Do(req)
-	require.NoError(t, err)
+	maxRetries := 5
```
Comment (bogdandrutu) [inline service/telemetry/metrics_test.go:216]:
```suggestion
	var rr  *http.Response
	maxRetries := 5
```

File: service/telemetry/metrics_test.go [hunk 3]
```diff
@@ -201,11 +202,31 @@ func createTestMetrics(t *testing.T, mp metric.MeterProvider) {
...
+	for i := 0; i < maxRetries; i++ {
+		rr, err = client.Do(req)
+		if err == nil && rr.StatusCode == http.StatusOK {
+			break
+		}
+
+		if i < maxRetries-1 {
```
Comment (bogdandrutu) [inline service/telemetry/metrics_test.go:223]: Add a small comment:

// skip sleep for last retry.

**Comment by jackgopack4** [comment]
thanks for the suggestions, applied them

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**